tests/id/id.cpp

```diff
@@ -284,16 +284,21 @@ TEMPLATE_TEST_CASE_SIG("id supports get() and operator[]", "[id]", ((int D), D),
   }
 }
 
-// FIXME: re-enable when sycl::id<>::dimensions is implemented
-// Issue link https://github.com/intel/llvm/issues/9786
-DISABLED_FOR_TEMPLATE_TEST_CASE_SIG(DPCPP, hipSYCL)
-("id provides static constexpr member 'dimensions'", "[id]", ((int D), D), 1, 2,
- 3)({
-  CHECK(std::is_same_v<decltype(sycl::id<D>::dimensions), const int>);
-  KCHECK(EVAL((std::is_same_v<decltype(sycl::id<D>::dimensions), const int>)));
-
-  CHECK(sycl::id<D>::dimensions == D);
-  KCHECK(EVAL(sycl::id<D>::dimensions) == D);
+DISABLED_FOR_TEST_CASE(hipSYCL)
+("id provides static constexpr member 'dimensions'", "[id]")({
+  CHECK(std::is_same_v<decltype(sycl::id<1>::dimensions), const int>);
+  CHECK(std::is_same_v<decltype(sycl::id<2>::dimensions), const int>);
+  CHECK(std::is_same_v<decltype(sycl::id<3>::dimensions), const int>);
+  KCHECK(EVAL((std::is_same_v<decltype(sycl::id<1>::dimensions), const int>)));
+  KCHECK(EVAL((std::is_same_v<decltype(sycl::id<2>::dimensions), const int>)));
+  KCHECK(EVAL((std::is_same_v<decltype(sycl::id<3>::dimensions), const int>)));
+
+  CHECK(sycl::id<1>::dimensions == 1);
+  CHECK(sycl::id<2>::dimensions == 2);
+  CHECK(sycl::id<3>::dimensions == 3);
+  KCHECK(EVAL(sycl::id<1>::dimensions) == 1);
+  KCHECK(EVAL(sycl::id<2>::dimensions) == 2);
+  KCHECK(EVAL(sycl::id<3>::dimensions) == 3);
```

Comment:
Note: The names given to the kernel in each instantiation of this test case would previously not be unique, hence the split here.

Reply:
This is interesting design information that should appear in a comment.

Reply:
Good point. I've added a comment about it.